frontend/pages/admin/IntegrationsPage/cards/MdmSettings/components/EndUserMigrationSection/EndUserMigrationSection.tsx

```diff
@@ -0,0 +1,217 @@
+import React, { useContext, useState } from "react";
+import { InjectedRouter } from "react-router";
+import classnames from "classnames";
+
+import PATHS from "router/paths";
+import configAPI from "services/entities/config";
+import { AppContext } from "context/app";
+import { NotificationContext } from "context/notification";
+
+// @ts-ignore
+import InputField from "components/forms/fields/InputField";
+import Radio from "components/forms/fields/Radio/Radio";
+import Slider from "components/forms/fields/Slider/Slider";
+import Button from "components/buttons/Button/Button";
+import validateUrl from "components/forms/validators/valid_url";
+import PremiumFeatureMessage from "components/PremiumFeatureMessage/PremiumFeatureMessage";
+import EmptyTable from "components/EmptyTable/EmptyTable";
+
+import ExampleWebhookUrlPayloadModal from "../ExampleWebhookUrlPayloadModal/ExampleWebhookUrlPayloadModal";
+
+const baseClass = "end-user-migration-section";
+
+const VOLUNTARY_MODE_DESCRIPTION =
+  "The end user sees the above window when they select Migrate to Fleet in the Fleet Desktop menu. If they’re unenrolled from your old MDM, the window appears every 15 minutes.";
+const FORCED_MODE_DESCRIPTION =
+  "The end user sees the above window every 15 minutes.";
+
+interface IEndUserMigrationFormData {
```

Comment:
I did this to distinguish the form client-side data from the API response data. I can see arguments for doing either in this case.

But my general thinking is it makes sense to have an interface for the API response data that is different than the client-side data. The reason is that at times we may want to represent the client-side data differently than the API response data. I see situations where we would want to change the structure of the data to better suit the UI. In this case that is not the case, but overall I think the pattern of separating the two is where we want to go. what do you think?

Reply:
I can definitely appreciate the thinking behind maintaining a distinction between them in cases like this. I guess I generally lean towards minimizing the number of interfaces where possible so that there's less surface area to maintain or refactor if things change down the road. But I'm ok either way.